packages/core/utils/lib/convert-query-params.js

```diff
@@ -174,7 +183,15 @@ const convertPopulateObject = (populate, schema) => {
     if (attribute.type === 'dynamiczone') {
       const populates = attribute.components
         .map((uid) => strapi.getModel(uid))
-        .map((schema) => convertNestedPopulate(subPopulate, schema));
+        .map((schema) => {
+          const populate = convertNestedPopulate(subPopulate, schema);
+          return isBoolean(populate) && populate ? {} : populate;
```

Comment:
It's because we are merging the populate later on and a boolean value would mess with it.

Reply:
we where already returning boolean values in some cases before though 🤔 was it a bug before ?

Reply:
Yes it was bug that never appeared I think :P

Reply:
then I would suggest

```suggestion
          return convertNestedPopulate(subPopulate, schema) || {}
```

Reply:
I don't think it would work as I need to remove the populate if it's `false` (let's say there is only one populate that is `false`, I want the `populates` object to be empty) and I need it to be an empty object if it's `true`.

I agree it not a really pretty code. I can do 2 maps instead of one maybe?